src/events.js

```diff
@@ -2308,53 +2322,89 @@ export const populateReportManifestTable = (currPage, searchSpecimenInstituteArr
-    let first = document.getElementById('firstPage');
-    let previous = document.getElementById('previousPage');
-    let current = document.getElementById('thisPage');
-    let next = document.getElementById('nextPage');
-    let final = document.getElementById('lastPage');
-    let middleNumber = document.getElementById('middlePage');
-    first.addEventListener('click', () => {
-        middleNumber.innerHTML = '1'
-        populateBoxTable(0, filter, source)
-    })
+    const firstEle = document.getElementById('firstPage');
+    const previousEle = document.getElementById('previousPage');
+    const nextEle = document.getElementById('nextPage');
+    const lastEle = document.getElementById('lastPage');
+    const currPageEle = document.getElementById('middlePage');
 
-    previous.addEventListener('click', () => {
-        middleNumber.innerHTML = middleNumber.innerHTML == '1' ? '1' : parseInt(middleNumber.innerHTML) - 1;
-        populateBoxTable(parseInt(middleNumber.innerHTML) - 1, filter, source)
-    })
+    // Update the current page number and the UI, then load the new page
+    const setPage = (targetPageNum) => {
+        const newPageNum = parseInt(targetPageNum, 10);
+        if (currPageNum === newPageNum) return;
 
-    next.addEventListener('click', () => {
-        middleNumber.innerHTML = parseInt(middleNumber.innerHTML) >= lastPage ? (lastPage == 0 ? 1 : lastPage.toString()) : parseInt(middleNumber.innerHTML) + 1;
-        populateBoxTable(parseInt(middleNumber.innerHTML) - 1, filter, source)
-    })
+        currPageNum = newPageNum;
+        currPageEle.innerHTML = currPageNum;
+        appState.setState({ reportPage: currPageNum, reportPageBoxData: null });
 
-    final.addEventListener('click', () => {
-        middleNumber.innerHTML = lastPage == 0 ? 1 : lastPage;
-        populateBoxTable(lastPage == 0 ? 0 : lastPage - 1, filter, source)
-    })
+        handleBoxReportsData(filter, source);
 
+    }
 
-}
+    firstEle.addEventListener('click', () => {
+        setPage(1)
+    });
 
-export const addEventFilter = (source) => {
+    previousEle.addEventListener('click', () => {
+        if (currPageNum > 1) setPage(currPageNum - 1);
+    });
+
+    nextEle.addEventListener('click', () => {
+        if (currPageNum < numReportPages) setPage(currPageNum + 1);
+    });
+
+    lastEle.addEventListener('click', () => {
+        setPage(numReportPages)
+    });
+
+    // Enable go to page feature.
+    document.getElementById('goToPageButton').addEventListener('click', () => {
+        const input = document.getElementById('goToPageInput');
+        const pageNumber = parseInt(input.value.trim(), 10);
+    
+        // Validate the input to ensure it's a valid page number within the range
+        if (!isNaN(pageNumber) && pageNumber >= 1 && pageNumber <= numReportPages) {
+            setPage(pageNumber);
+        } else {
+            alert(`Please enter a valid page number (1 - ${numReportPages}).`);
```

Comment:
Should we display last page when input value is larger than numReportPages, and first page when input value <= 0 ?

Reply:
I like this suggestion, thanks. Updated. 

Just to be sure we're on the same page, I have this 1-indexed to match the page number the user sees, so anything < 1 goes to page 1.